test/test_sparse_prop.py

```diff
@@ -0,0 +1,97 @@
+# Owner(s): ["module: sparse"]
+
+import torch
+
+from torch.testing._internal.common_utils import TestCase, run_tests
+from typing import Optional, Tuple
+
+
+class SumNet(torch.nn.Module):
+    def __init__(self):
+        super(SumNet, self).__init__()
+        return
+
+    def forward(self, x):
+        return x.sum()
+
+
+class EltwiseNet(torch.nn.Module):
+    def __init__(self):
+        super(EltwiseNet, self).__init__()
+        return
+
+    def forward(self, x):
+        return 2 * torch.sin(-x)
+
+
+# All sparse layouts currently supported in torch.sparse.
+# TODO: make this work for all
+SPARSE_LAYOUTS = [
+    torch.sparse_coo,
+    #    torch.sparse_csr,
+    #    torch.sparse_csc,
+    #    torch.sparse_bsr,
+    #    torch.sparse_bsc,
+]
+
+
+def make_sparse(layout: torch.layout) -> Tuple[torch.Tensor, Optional[Tuple[int, int]]]:
+    dense = torch.ones(4, 8)
+    if layout == torch.sparse_coo:
+        return (dense.to_sparse_coo(), None)
+    if layout == torch.sparse_csr:
+        return (dense.to_sparse_csr(), None)
+    if layout == torch.sparse_csc:
+        return (dense.to_sparse_csc(), None)
+    if layout == torch.sparse_bsr:
+        return (dense.to_sparse_bsr((2, 2)), (2, 2))
+    if layout == torch.sparse_bsc:
+        return (dense.to_sparse_bsc((2, 4)), (2, 4))
+
+
+class TestSparseProp(TestCase):
+    def setUp(self):
+        TestCase.setUp(self)
+
+    def test_sumnet(self):
+        net = SumNet()
+        for sparse_layout in SPARSE_LAYOUTS:
```

Comment:
See TestSparseAny in test/test_sparse.py how we typically write tests for different layouts. Basically, use
```suggestion
    @all_sparse_layouts('layout', include_strided=False)
    def test_sumnet(self, layout):
        net = SumNet()
        if layout not in SPARSE_LAYOUTS:
            self.skipTest(f'layout support not yet implemented')
```

Reply:
Oh, that is nice, yes. I like testing constructs like that.